modules/gapi/include/opencv2/gapi/gkernel.hpp

```diff
@@ -199,16 +207,17 @@ class GKernelTypeM<K, std::function<std::tuple<R...>(Args...)> >:
 template<typename, typename> class GKernelType;
 
 template<typename K, typename R, typename... Args>
-class GKernelType<K, std::function<R(Args...)> >:
-        public detail::MetaHelper<K, std::tuple<Args...>, R>
+class GKernelType<K, std::function<R(Args...)> >
```

Comment:
That FIXME is there since late 2017.

Reply:
And actually that `GKernelTypeMediator` thing solves exactly the same problem (for inference API)

Reply:
Oh, I see the FIXME.

`GKernelTypeMediator` is a good thing but it introduces additional abstraction (increases number of code lines). My proposal is to drop one of the versions altogether and have unified logic (thus reducing the number of code but preserving the same functionality). I'm OK to do it separately and since there's a FIXME already, consider this discussion resolved from my side